### DISCUSSION TITLE
Session Browser

### DISCUSSION THREAD

**Discussion by joshjody** [fastapi/fastapi#3645] [Questions] [answered]
how to get or set session browser in fastapi like in flask to get session "session.get('username')" or set session "session['username']     = username"?

**Comment by Cosmicoppai** [comment]
You can use OAuth with JWT.
Hope this will help:- https://fastapi.tiangolo.com/tutorial/security/get-current-user/
https://fastapi.tiangolo.com/tutorial/security/oauth2-jwt/

**Comment by AlexandreBiguet** [comment, marked as answer]
Hello :) 

Perhaps the starlette [session middleware](https://www.starlette.io/middleware/#sessionmiddleware) is what you need here ? 

Sample code (not tested) that should show basic usage:

```python
from fastapi import FastAPI
from fastapi.responses import HTMLResponse, RedirectResponse

from starlette.requests import Request
from starlette.middleware.sessions import SessionMiddleware


app = FastAPI()
app.add_middleware(SessionMiddleware, secret_key="my-secret")


@app.get("/")
def home(request: Request):
    data = request.session.get("my_data")
    if data:
        html = f"<pre>{data}</pre>" '<a href="/data_erase"> erase some data</a>'
        return HTMLResponse(html)
    return HTMLResponse('<a href="/data_set"> set some data</a>')


@app.get("/data_set")
def data_set(request: Request):
    request.session["my_data"] = {"hello": ", world"}
    return RedirectResponse(url="/")


@app.get("/data_erase")
def data_erase(request: Request):
    request.session.pop("my_data")
    return RedirectResponse(url="/")
```

Hope this helps, 

Cheers

**Reply by AlexandreBiguet** [reply]
BTW I just came across this issue: https://github.com/tiangolo/fastapi/issues/754 which I think you should have a look at.

Cheers
